src/main/java/com/cognifide/aemrules/checks/UnsychronizedModificationCheck.java

```diff
@@ -0,0 +1,134 @@
+package com.cognifide.aemrules.checks;
+
+import java.util.Deque;
+import java.util.List;
+import java.util.Set;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.SubscriptionBaseVisitor;
+import org.sonar.java.model.ModifiersUtils;
+import org.sonar.java.tag.Tag;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.AssignmentExpressionTree;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.ExpressionStatementTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.Modifier;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import com.cognifide.aemrules.checks.visitors.AnnotationsVisitor;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+@Rule(
+		key = "AEM-13",
+		name = "Modification of a Service fields should be synchronized.",
+		priority = Priority.CRITICAL,
+		tags = {Tag.BUG, Tag.MULTI_THREADING}
+)
+public class UnsychronizedModificationCheck extends SubscriptionBaseVisitor {
+
+	private Deque<Boolean> inSynchronizedBlock = Lists.newLinkedList();
+
+	private Set<Symbol> fields = Sets.newHashSet();
+
+	private boolean service;
+
+	@Override
+	public void scanFile(JavaFileScannerContext context) {
+		inSynchronizedBlock.push(false);
+		service = true;
```

Comment:
setting the value in this place is a little bit confusing. It would be better to move it to filed declaration